addons/project/models/project.py

```diff
@@ -1500,7 +1521,16 @@ def copy(self, default=None):
             default['recurrence_id'] = self.recurrence_id.copy().id
         if self.allow_subtasks:
             default['child_ids'] = [child.copy().id for child in self.child_ids]
-        return super(Task, self).copy(default)
+
+        new_task = super(Task, self).copy(default)
+
+        if self.allow_task_dependencies:
+            for depend_on_id in self.depend_on_ids:
+                new_dependent_ids = [dependent_id.id for dependent_id in depend_on_id.dependent_ids]
+                new_dependent_ids.append(new_task.id)
+                self.env['project.task'].browse(depend_on_id.id).write({'dependent_ids': new_dependent_ids})
```

Comment:
It is not equivalent to:
```suggestion
            self.depend_on_ids.write({'dependent_ids': [Command.link(new_task.id)]})
```
? 🤔

Reply:
i didn't know about its existance when doing this task, thank you